BossMod/Data/ClientState.cs

```diff
@@ -50,16 +50,28 @@ public record struct Stats(int SkillSpeed, int SpellSpeed, int Haste);
     public Stats PlayerStats;
     public readonly Cooldown[] Cooldowns = new Cooldown[NumCooldownGroups];
     public readonly ActionID[] DutyActions = new ActionID[2];
+    public readonly byte[] DutyActionCharges = new byte[2];
     public readonly byte[] BozjaHolster = new byte[(int)BozjaHolsterID.Count]; // number of copies in holster per item
+    public readonly uint[] BlueMageSpells = new uint[24];
     public readonly short[] ClassJobLevels = new short[NumClassLevels];
     public Fate ActiveFate;
+    public Pet ActivePet;
 
     public int ClassJobLevel(Class c)
     {
         var index = Service.LuminaRow<Lumina.Excel.GeneratedSheets.ClassJob>((uint)c)?.ExpArrayIndex ?? -1;
         return index >= 0 && index < ClassJobLevels.Length ? ClassJobLevels[index] : -1;
     }
 
+    public unsafe T GetGauge<T>() where T : unmanaged
```

Comment:
Why is it needed here? This is a very ugly hack...

Reply:
i put this in sch rotation:

```c#
        d.Spell(AID.EnergyDrain)!.ForbidExecute =
            (ws, _, _, _) => ws.Client.GetGauge<ScholarGauge>().Aetherflow == 0;
```

i could move it somewhere else i guess, but it's still going to involve an unsafe cast

Reply:
I see, ok, yeah it makes sense...